### ISSUE TITLE
Is it possible to slightly manipulate the output HTML?

### ISSUE THREAD

**Issue by sindoc** [ericchiang/pup#187]
Is it possible to use pup to add HTML elements under a particular node?

Say I need to add <link rel="shortcut icon" href="/favicon.ico"> under 'html > head'

**Comment by gromgit** [comment]
No. `pup` only selects and outputs existing HTML. It's also pretty much abandoned, so I wouldn't hope for enhancements at this point.

You probably need something like [XMLStarlet](https://xmlstar.sourceforge.net/) instead. It's even older than `pup`, and it's really intended for XML manipulation, but it should work OK with HTML that's not completely sloppy.
